.github/actions/release-notes/release_notes_cli.py

```diff
@@ -253,12 +258,19 @@ def github_api_lookup(repo_url):
 
     if parsed.local_release_notes:
         with open(parsed.local_release_notes, 'w') as f:
-            f.write(release_notes_md)
+            f.write(release_notes_docs_markdown)
 
     if parsed.subcomponent_release_notes:
         with open(parsed.subcomponent_release_notes, 'w') as f:
             f.write(sub_component_release_notes)
 
+    if parsed.tar_output:
+        tar_stream = b''.join(rnt.release_notes_docs_into_tarstream(
+            release_notes_docs + grouped_sub_component_release_notes_docs
+        ))
+        with open(parsed.tar_output, 'wb') as f:
+            f.write(tar_stream)
```

Comment:
I suggest to use streaming, considering we get a generator in the first place:

```
  for chunk in ...:
    f.write(chunk)
```

It will very likely not be an issue to read release-notes into memory, it is not a nice pattern to do so IMO.